### DISCUSSION TITLE
add unsigned 64 bit integer to base model

### DISCUSSION THREAD

**Discussion by pahjbo** [ivoa/vo-dml#57] [Ideas]
The base model only deals with integers - it is probably useful to add the unsigned integer distinction, and if so then it might be necessary to add size.

**Comment by pdowler** [comment]
The current spec does not specify the size of real (float or double or more) or integer (byte,short,int,long,...), so they are intentionally more abstract than an implementation. I don't recall the reasoning behind that and there are consequences when you try to have two implementations that "interoperate" in the nitty gritty details.

**Comment by pahjbo** [comment]
this is here as the placeholder for a discussion going on in the P3t mailing list - the title is probably misleading -  it actually should probably be more general in that if we want to ensure future interoperability of binary representations of the model instances then we might have to say something about the binary representations of the various primitives.
